vm/src/obj/objbytes.rs

```diff
@@ -269,6 +268,28 @@ impl PyBytesRef {
         obj => {Err(vm.new_type_error(format!("{} cannot be interpreted as an integer", obj)))}
         )
     }
+
+    #[pymethod(name = "count")]
+    fn count(
+        self,
+        sub: PyObjectRef,
+        start: OptionalArg<PyObjectRef>,
+        end: OptionalArg<PyObjectRef>,
+        vm: &VirtualMachine,
+    ) -> PyResult {
+        match is_bytes_like(&sub) {
```

Comment:
I suggest to move this logic about sub being an integer value into the inner object, so we can re-use it for `bytearray`.